src/consumption.cpp

```diff
@@ -1095,6 +976,134 @@ void Character::modify_morale( item &food, const int nutr )
         add_morale( MORALE_FOOD_GOOD, fun.first, fun.second, morale_time, morale_time / 2, false,
                     food.type );
     }
+
+    bool spoiled = food.rotten();
```

Comment:
This better be inlined. `spoiled` isn't any better than `food.rotten()`.

Reply:
Resolved in https://github.com/CleverRaven/Cataclysm-DDA/pull/37713/commits/0f5b15d7820aa0ecb1df51b494fea45153ddaad9